### ISSUE TITLE
Can't install Ruby 2.3.0 on OS X 10.11.4

### ISSUE THREAD

**Issue by jonathansimmons** [rvm/rvm#3671]
I'm running into an install issue for Ruby 2.3.0 and can't discern what exactly the issue is. Here is the output of the error: 

```bash
$ rvm install ruby-2.3.0
ruby-2.3.0 - #removing src/ruby-2.3.0..
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.3.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Warning: DYLD_LIBRARY_PATH environment variable is set, this might interact with the compilation and ruby.
Installing Ruby from source to: /Users/jonathan.simmons/.rvm/rubies/ruby-2.3.0, this may take a while depending on your cpu(s)...
ruby-2.3.0 - #downloading ruby-2.3.0, this may take a while depending on your connection...
ruby-2.3.0 - #extracting ruby-2.3.0 to /Users/jonathan.simmons/.rvm/src/ruby-2.3.0....
ruby-2.3.0 - #configuring...........................................................
ruby-2.3.0 - #post-configuration.
ruby-2.3.0 - #compiling.............................................................
ruby-2.3.0 - #installing......
Error running '__rvm_make install',
showing last 15 lines of /Users/jonathan.simmons/.rvm/log/1461772802_ruby-2.3.0/install.log
installing default psych libraries
installing default pty libraries
installing default ripper libraries
checking ../.././parse.y and ../.././ext/ripper/eventids2.c
linking shared-object socket.bundle
installing default socket libraries
installing default syslog libraries
installing tcltklib libraries
installing default tcltklib libraries
make[2]: `ruby' is up to date.
./miniruby -I./lib -I. -I.ext/common  ./tool/runruby.rb --extout=.ext  -- --disable-gems -r./x86_64-darwin15-fake ./tool/rbinstall.rb --make="/Applications/Xcode.app/Contents/Developer/usr/bin/make" --dest-dir="" --extout=".ext" --mflags="" --make-flags="" --data-mode=0644 --prog-mode=0755 --installed-list .installed.list --mantype="doc"
./tool/runruby.rb:106:in `exec': Input/output error - ./ruby (Errno::EIO)
	from ./tool/runruby.rb:106:in `<main>'
make: *** [do-install-nodoc] Error 1
++ return 2
There has been an error while running make install. Halting the installation.
```

Any help is appreciated! 👍

**Comment by jonathansimmons** [comment]
@rys @wayneeseguin @mpapis any ideas?


Results of `env | sort`

```
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.HR6RcTMe3m/Render
GEM_HOME=/Users/jonathan.simmons/.rvm/gems/ruby-2.2.3
GEM_PATH=/Users/jonathan.simmons/.rvm/gems/ruby-2.2.3:/Users/jonathan.simmons/.rvm/gems/ruby-2.2.3@global
HOME=/Users/jonathan.simmons
IRBRC=/Users/jonathan.simmons/.rvm/rubies/ruby-2.2.3/.irbrc
LANG=en_US.UTF-8
LOGNAME=jonathan.simmons
MEMDEF_POLICY=DDADDDDDDDDDDAAAAAAAADD:C
MY_RUBY_HOME=/Users/jonathan.simmons/.rvm/rubies/ruby-2.2.3
OLDPWD=/Users/jonathan.simmons/git/gateway
PATH=/Users/jonathan.simmons/.rvm/gems/ruby-2.2.3/bin:/Users/jonathan.simmons/.rvm/gems/ruby-2.2.3@global/bin:/Users/jonathan.simmons/.rvm/rubies/ruby-2.2.3/bin:/Users/jonathan.simmons/.rvm/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/mysql/bin
PS1=\u@\h \[\033[32m\]\w$(parse_git_branch)\[\033[00m\] $ 
PWD=/Users/jonathan.simmons/git/gateway/time-dot
RUBY_VERSION=ruby-2.2.3
SECURITYSESSIONID=186a9
SHELL=/bin/bash
SHLVL=1
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.aFFJyC7vB2/Listeners
TERM=xterm-256color
TERM_PROGRAM=Apple_Terminal
TERM_PROGRAM_VERSION=361.1
TERM_SESSION_ID=5CBDC3FC-96B2-4023-8B2F-8B405028D953
TMPDIR=/var/folders/bt/b2_t07j1213_d1kw80twgqwjg9c9b8/T/
USER=jonathan.simmons
XPC_FLAGS=0x0
XPC_SERVICE_NAME=0
_=/usr/bin/env
__CF_USER_TEXT_ENCODING=0x5E962568:0x0:0x0
_system_arch=x86_64
_system_name=OSX
_system_type=Darwin
_system_version=10.11
rvm_bin_path=/Users/jonathan.simmons/.rvm/bin
rvm_delete_flag=0
rvm_path=/Users/jonathan.simmons/.rvm
rvm_prefix=/Users/jonathan.simmons
rvm_ruby_string=ruby-2.2.3
rvm_version=1.27.0 (master)
```

**Comment by mpapis** [comment]
I wonder about `DYLD_LIBRARY_PATH` warning, it does not show up in the `env` - yet rvm detected it, can you try: `echo $DYLD_LIBRARY_PATH`? maybe you have in some configuration file?

**Comment by jonathansimmons** [comment]
That was a fix for an old mysql issue. It's in my `.bash_profile` but even taking that out I run into the same issue. Which is sort of not surprising as it was never a problem before. 

it looks like there error is more specific to: 

`/tool/runruby.rb:106:in `exec': Input/output error - ./ruby (Errno::EIO)` but I don't know where to start and would probably be in over my head. 

Are any others able to 2.3.0 on this version of OS X without an issue?

**Comment by mpapis** [comment]
hmm - `make[2]: `ruby' is up to date.` - try full reinstall:
```bash
rvmsudo rvm remove ruby-2.3.0
rvm install ruby-2.3.0
```

**Comment by jonathansimmons** [comment]
The `rvmsudo` remove worked! Not sure what went wrong to begin with. Thanks so much for the input!

**Comment by painofblaine19** [comment]
I'm having the same issue, but when I use the rvmsudo, I get this: 

    Jacobs-MacBook-Pro:~ jacob$ rvmsudo rvm remove ruby-2.0.0.p648
    Warning: can not check `/etc/sudoers` for `secure_path`, falling back to call via `/usr/bin/env`, this breaks rules from `/etc/sudoers`. Run:

        export rvmsudo_secure_path=1

    to avoid the warning, put it in shell initialization file to make it persistent.

    In case there is no `secure_path` in `/etc/sudoers`. Run:

        export rvmsudo_secure_path=0

    to avoid the warning, put it in shell initialization file to make it persistent.

**Comment by mpapis** [comment]
@painofblaine19 this is just a warning, read it, read about `secure_path` in `/etc/sudoers` - and decide how you would prefer to silence this warning

**Comment by pkuczynski** [comment]
We just uploaded 2.3.1 binaries for OSX El Capitan.

```
rvm get master
rvm install 2.3.1
```

should solve your problem.

**Comment by Loremaster** [comment]
@pkuczynski I don't see any binaries upload for ruby 2.3 here: https://rvm.io/binaries/osx/10.11/x86_64/

Is it possible to add that? Because running:

```
$ rvm get master
$ rvm install 2.3.1
```

Gives me that:

```
Warning! PATH is not properly set up, '/Users/serj/.rvm/gems/ruby-2.2.1@email_platform/bin' is not at first place,
         usually this is caused by shell initialization files - check them for 'PATH=...' entries,
         it might also help to re-add RVM to your dotfiles: 'rvm get stable --auto-dotfiles',
         to fix temporarily in this shell session run: 'rvm use ruby-2.2.1@email_platform'.
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Warning: Use homebrew/dupes/apple-gcc42 instead of deprecated homebrew/core/apple-gcc42
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Warning: found user selected compiler '/usr/bin/gcc', this will suppress RVM auto detection mechanisms.
Installing Ruby from source to: /Users/serj/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0   943k      0  0:00:14  0:00:14 --:--:--  995k
ruby-2.3.1 - #extracting ruby-2.3.1 to /Users/serj/.rvm/src/ruby-2.3.1....
ruby-2.3.1 - #configuring...........................................................
ruby-2.3.1 - #post-configuration.
ruby-2.3.1 - #compiling............................................
Error running '__rvm_make -j 1',
showing last 15 lines of /Users/serj/.rvm/log/1478727564_ruby-2.3.1/make.log
                                      ^
etc.c:788:56: error: passing 'void' to parameter of incompatible type 'VALUE' (aka 'unsigned long')
    rb_hash_aset(result, ID2SYM(rb_intern("machine")), rb_str_new_cstr(u.machine));
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~
../.././include/ruby/intern.h:803:30: note: expanded from macro 'rb_str_new_cstr'
#define rb_str_new_cstr(str) __extension__ (    \
                             ^~~~~~~~~~~~~~~~~~~~
../.././include/ruby/intern.h:520:39: note: passing argument to parameter here
VALUE rb_hash_aset(VALUE, VALUE, VALUE);
                                      ^
1 warning and 6 errors generated.
make[2]: *** [etc.o] Error 1
make[1]: *** [ext/etc/all] Error 2
make: *** [build-ext] Error 2
++ return 2
There has been an error while running make. Halting the installation.
```

so basically it doesn't download binaries.

**Comment by pkuczynski** [comment]
Please gist full output of `/Users/serj/.rvm/log/1478727564_ruby-2.3.1/make.log`

**Comment by IvRRimum** [comment]
I am affected by the same problem.

**Comment by IvRRimum** [comment]
```
karliscudars@Karliss-MacBook-Pro:~$ rvm install 2.3.1
ruby-2.3.1 - #removing src/ruby-2.3.1..
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.3.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Installing requirements for osx.
Updating system.....
Installing required packages: coreutils...
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/karliscudars/.rvm/rubies/ruby-2.3.1, this may take a while depending on your cpu(s)...
ruby-2.3.1 - #downloading ruby-2.3.1, this may take a while depending on your connection...
ruby-2.3.1 - #extracting ruby-2.3.1 to /Users/karliscudars/.rvm/src/ruby-2.3.1....
ruby-2.3.1 - #configuring......
Error running './configure --prefix=/Users/karliscudars/.rvm/rubies/ruby-2.3.1 --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/readline:/usr/local/opt/libksba:/usr/local/opt/openssl --disable-install-doc --enable-shared',
showing last 15 lines of /Users/karliscudars/.rvm/log/1478952587_ruby-2.3.1/configure.log
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for cd using physical directory... cd -P
checking whether CFLAGS is valid... yes
checking whether LDFLAGS is valid... no
configure: error: something wrong with LDFLAGS=""
There has been an error while running configure. Halting the installation.
karliscudars@Karliss-MacBook-Pro:~$ 
```

**Comment by IvRRimum** [comment]
False alarm. My ~/.bash_profile contained LDFLAGS export. Removing it, solved the issue.
